[toolchain] uClibc: fix errno access in pthreads (#8166, #8177)
Reading errno within pthreads always resulted in 0 breaking various programs
like transmission and collectd. This patch is not 100% thread safe but it brings
back threading into a usable state for now.

--- a/libpthread/linuxthreads.old/errno.c
+++ b/libpthread/linuxthreads.old/errno.c
@@ -22,6 +22,7 @@
 #include "internals.h"
 #include <stdio.h>
 
+libpthread_hidden_proto(__errno_location)
 int *
 __errno_location (void)
 {
@@ -29,6 +30,7 @@ __errno_location (void)
   return THREAD_GETMEM (self, p_errnop);
 }
 
+libpthread_hidden_proto(__h_errno_location)
 int *
 __h_errno_location (void)
 {
